# I. Getting the Dataset

In [1]:
# #!/bin/bash
# mkdir -p ./ds/
# curl -L -o ./ds/7k-books-with-metadata.zip https://www.kaggle.com/api/v1/datasets/download/dylanjcastillo/7k-books-with-metadata

In [1]:
import pandas as pd

In [5]:
books = pd.read_csv("./ds/books.csv")
books.sample(5)

isbn13      isbn10                          title  \
5673  9781101098394  1101098392  The Hound of the Baskervilles   
2818  9780440910817  0440910811        A Ring of Endless Light   
2684  9780439411370  0439411378                     The Summit   
181   9780060554736  0060554738                       The Game   
5405  9780859653725  0859653722      Dissecting Marilyn Manson   

                                              subtitle  \
5673                                               NaN   
2818                                               NaN   
2684                                               NaN   
181   Penetrating the Secret Society of Pickup Artists   
5405                                               NaN   

                     authors                 categories  \
5673  Sir Arthur Conan Doyle                    Fiction   
2818       Madeleine L'Engle                      Death   
2684           Gordon Korman           Juvenile Fiction   
181             Neil Strauss  Biography & Autobiography   
5405          Gavin Baddeley                      Music   

                                              thumbnail  \
5673  http://books.google.com/books/content?id=rtBSj...   
2818  http://books.google.com/books/content?id=UGoVw...   
2684  http://books.google.com/books/content?id=6VlnC...   
181   http://books.google.com/books/content?id=ibpvw...   
5405  http://books.google.com/books/content?id=7_FrA...   

                                            description  published_year  \
5673  Sherlock Holmes and Dr. Watson face a mystery ...          2001.0   
2818  During the summer her grandfather is dying of ...          1995.0   
2684  As Tilt, Perry, Sammi, and Dominic prepare to ...          2002.0   
181   Hidden somewhere, in nearly every major city i...          2005.0   
5405  Marilyn Manson in more than a mere rock star -...          2005.0   

      average_rating  num_pages  ratings_count  
5673             NaN        NaN            NaN  
2818            4.15      332.0        16513.0  
2684            3.96      154.0         2515.0  
181             3.74      464.0        19305.0  
5405            3.86      192.0          504.0

# 1. EDA

## 1.1 Checking if there are null values in the DS

In [24]:
books.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6810 entries, 0 to 6809
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   isbn13          6810 non-null   int64  
 1   isbn10          6810 non-null   object 
 2   title           6810 non-null   object 
 3   subtitle        2381 non-null   object 
 4   authors         6738 non-null   object 
 5   categories      6711 non-null   object 
 6   thumbnail       6481 non-null   object 
 7   description     6548 non-null   object 
 8   published_year  6804 non-null   float64
 9   average_rating  6767 non-null   float64
 10  num_pages       6767 non-null   float64
 11  ratings_count   6767 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 638.6+ KB


In [9]:
null_counts = books.isnull().sum()
null_counts

isbn13               0
isbn10               0
title                0
subtitle          4429
authors             72
categories          99
thumbnail          329
description        262
published_year       6
average_rating      43
num_pages           43
ratings_count       43
dtype: int64

### What we inferred?
- 6810 entries in total
- isbn ids and titles are all fine
- a lot of subtitles are missing, approx 2/3 of total
- 
